### PR TITLE
Add non-mod libraries guide (libraries.ytag)

### PR DESCRIPTION
**Pull request by Leo40Git**
Just a simple guide to what shadowing is and what it's useful for.
Most instructions are offloaded to the plugin's doc site.

### CODE REVIEW

**Review by Scotsguy — APPROVED** [review]
There's some disagreement over whether to use shadow or just jij libraries, but shadow is probably the "cleaner" way anways

File: tags/guide/shadow.ytag [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          inline: false
+          value: A recommended way to include a dependency that is **not** a mod (I.E a generic Java library) is to _shadow_ it.
+
+        - name: ...and what does that mean?
+          inline: false
+          value: _Shadowing_ simply means taking a library and including it in your distribution of the thing using said library (in this case, your mod).
+
+        - name: ...and how do I do that?
+          inline: false
+          value: Simple, you read the docs! https://imperceptiblethoughts.com/shadow/introduction/
```
Comment (i509VCB) [inline tags/guide/shadow.ytag:18]: I believe most people asking about shadow would like to shadow stuff into their mods. Ideally a solution to that would be nice but is not immediately urgent.

**Review by i509VCB — COMMENTED** [review]
I do believe this tag could describe the recommended cases in which shadowing should be used (`??jij` for example)

File: tags/guide/shadow.ytag (hunk 1 above)
Comment (Leo40Git) [reply tags/guide/shadow.ytag]: Added a short little "crash course" of sorts about `shadow`ing.

**Comment by Leo40Git** [comment]
This tag now lists general information on including non-mod libraries, including whether to use `include` or `shadow`.

**Comment by JamiesWhiteShirt** [comment]
Shadowing HIGHLY INCREASES chances of conflicts if you don't relocate the external library's package, due to the way different versions of library may interact if they exist in the same package. [The gradle plugin doesn't do relocation by default.](https://imperceptiblethoughts.com/shadow/configuration/relocation/) In its current state, the tag appears to be completely oblivious to this fact, which really must be mentioned.

Shadow and Loom also don't integrate out of the box. Some configuration is required.

**Comment by Leo40Git** [comment]
Will add relocation configuration to the tag.

> Shadow and Loom also don't integrate out of the box. Some configuration is required.

More configuration than just adding the shadow plugin to build.gradle? Because that's already in the tag.

**Comment by JamiesWhiteShirt** [comment]
> More configuration than just adding the shadow plugin to build.gradle? Because that's already in the tag.

Yup - I don't know what the idiomatic solution currently is, but I've seen people mention specifying `shadowJar` as the input task for the `remapJar` task.

**Comment by Leo40Git** [comment]
Funky integration code by `HalfOf2#0086` (thanks, man!)

**Review by user11681 — APPROVED** [review]

**Comment by i509VCB** [comment]
I think the shadow section looks fine - future PRs can fix this if needed.

**Review by i509VCB — APPROVED** [review]

**Comment by i509VCB** [comment]
Hmm this doesn't seem to display - may need to look into that

**Comment by Leo40Git** [comment]
how do I even check what's going wrong

**Comment by YTG1234** [comment]
ask for bot logs?

Also, my `faq/launchers` hasn't been working as well

**Pull request merged**